### PR TITLE
Implement Mongo storage

### PR DESCRIPTION
**Pull request by devgony**
## Goal
- Implement Mongo storage to query by SQL
  
## Todo
- [x] Implement validator with jsonSchema for schema table
- [x] Implement data coversion between Value and Bson
- [x] Add example for unsupported types
  - [x] Regular Expression
  - [x] Min key
  - [x] Max key 
  - [x] JavaScript
  - [x] JavaScript code with scope~~
  - [x] deprecated: DBPointer, Symbol, Undefined

### CODE REVIEW

File: core/src/executor/insert.rs [hunk 1]
```diff
@@ -190,6 +190,7 @@ async fn fetch_map_rows<T: GStore>(storage: &T, source: &Query) -> Result<Vec<Da
             let limit = Limit::new(source.limit.as_ref(), source.offset.as_ref()).await?;
             let rows = stream::iter(values_list).then(|values| async move {
                 if values.len() > 1 {
+                    println!("{values:#?}");
```
Comment (ever0de) [inline core/src/executor/insert.rs:193]:
```suggestion
```

File: storages/mongo-storage/src/row/data_type.rs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            DataType::Uint8 => BsonType::Int32,
+            DataType::Uint16 => BsonType::Int32,
+            DataType::Uint32 => BsonType::Int64,
+            DataType::Uint64 => BsonType::Decimal128,
+            DataType::Uint128 => BsonType::Decimal128,
+            DataType::Float32 => BsonType::Double,
+            DataType::Float => BsonType::Double,
+            DataType::Text => BsonType::String,
+            DataType::Bytea => BsonType::Binary,
+            // DataType::Inet => BsonType::String,
```
Comment (ever0de) [inline storages/mongo-storage/src/row/data_type.rs:70]:
```suggestion
```

File: storages/mongo-storage/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+bson = { version = "2.6.1", features = ["chrono-0_4"] }
+chrono = { version = "0.4.26", features = ["serde", "wasmbind"] }
+strum_macros = "0.24"
+strum = "0.24"
+rust_decimal = { version = "1", features = ["serde-str"] }
+
+[dev-dependencies]
+test-suite.workspace = true
+uuid = "1"
+composite-storage.workspace = true
```
Comment (panarch) [inline storages/mongo-storage/Cargo.toml:37]: we can remove and rearrange unused and testonly deps. then
```toml
[dependencies]
gluesql-core.workspace = true

async-trait = "0.1"
futures = "0.3"
thiserror = "1.0"
mongodb = "2.5.0"
bson = { version = "2.6.1", features = ["chrono-0_4"] }
chrono = { version = "0.4.26", features = ["serde", "wasmbind"] }
strum_macros = "0.24"
strum = "0.24"
rust_decimal = { version = "1", features = ["serde-str"] }

[dev-dependencies]
test-suite.workspace = true

tokio = { version = "1", features = ["rt", "macros"] }
```

**Review by panarch — APPROVED** [review]
it's pleasant to ship this to the main. looks great!! thanks a lot for the contribution.

**Pull request merged**